superset/hive_query.py

```diff
@@ -39,72 +34,123 @@
     'P1Y'  : 31556926
 }
 
-# return seconds from gran valaue map
-def getGranValueInSeconds(value):
-    if  value in GRAN_VALUE_MAP:
+# return seconds from gran valaue map default is 86400
+def get_gran_value_in_seconds(value,time_partitions):
+    partition_min_grain = get_partitions_min_grain(time_partitions)
+    if value in GRAN_VALUE_MAP and  GRAN_VALUE_MAP[value] >= partition_min_grain:
         return GRAN_VALUE_MAP[value]
+    return partition_min_grain
+
+def get_partitions_min_grain(time_partitions):
```

Comment:
We should replace 'grain' with 'gran' everywhere

Reply:
superset using  grain key for time grain related naming conventions and replacing
GRAN_VALUE_MAP with GRAIN_VALUE_MAP